gevent/_socket2.py

```diff
@@ -334,10 +334,10 @@ def settimeout(self, howlong):
             howlong = f()
             if howlong < 0.0:
                 raise ValueError('Timeout value out of range')
-        self.timeout = howlong
+        self.__dict__['timeout'] = howlong # avoid recursion with any property on self.timeout
```

Comment:
This only becomes a problem when we turn SSLSocket.timeout into a property attribute. Maybe we can avoid that?